akka-testkit/src/main/scala/akka/testkit/scaladsl/SerializerTestKit.scala

```diff
@@ -0,0 +1,225 @@
+ * include the generated files in your VCS so that earlier versions of the serialized data is available for
+ * verification and incompatibilities can be detected. By default the data is expected to live in `src/test/serializer-testkit`
+ */
+object SerializerTestKit {
+  def apply()(implicit system: ActorSystem) = new SerializerTestKit(SerializerTestKitSettings(), None)
+  def apply(settings: SerializerTestKitSettings)(implicit system: ActorSystem) = new SerializerTestKit(settings, None)
+  def apply(serializer: ExtendedActorSystem ⇒ Serializer)(implicit system: ActorSystem) =
+    new SerializerTestKit(SerializerTestKitSettings(), Some(serializer(system.asInstanceOf[ExtendedActorSystem])))
+  def apply(settings: SerializerTestKitSettings, serializer: ExtendedActorSystem ⇒ Serializer)(implicit system: ActorSystem) =
+    new SerializerTestKit(settings, Some(serializer(system.asInstanceOf[ExtendedActorSystem])))
+
+  // minimal dummy implementation allowing for serialization like the actorref was realy but nothing else
+  private final case class StableDummyActorRef(override val path: ActorPath)(implicit _system: ActorSystem) extends ActorRefWithCell with ActorRefScope {
+    def system: ActorSystem = _system
+    val underlying: Cell = new Cell {
+      def parent: InternalActorRef = ???
+      def getChildByName(name: String): Option[ChildStats] = ???
+      private[akka] def isTerminated = ???
+      def sendSystemMessage(msg: SystemMessage): Unit = ???
+      def hasMessages: Boolean = ???
+      def numberOfMessages: Int = ???
+      def resume(causedByFailure: Throwable): Unit = ???
+      def suspend(): Unit = ???
+      def systemImpl: ActorSystemImpl = ???
+      def restart(cause: Throwable): Unit = ???
+      def start(): this.type = ???
+      def sendMessage(msg: Envelope): Unit = ???
+      def props: Props = ???
+      def isLocal: Boolean = ???
+      def getSingleChild(name: String): InternalActorRef = ???
+      def system: ActorSystem = _system
+      def stop(): Unit = ???
+      def self: ActorRef = ???
+      def childrenRefs: ChildrenContainer = ???
+    }
+    def children: immutable.Iterable[ActorRef] = ???
+    def getSingleChild(name: String): InternalActorRef = ???
+    def start(): Unit = ???
+    def suspend(): Unit = ???
+    def restart(cause: Throwable): Unit = ???
+    def resume(causedByFailure: Throwable): Unit = ???
+    def stop(): Unit = ???
+    def sendSystemMessage(message: SystemMessage): Unit = ???
+    def provider: ActorRefProvider = ???
+    def getParent: InternalActorRef = ???
+    def getChild(name: Iterator[String]): InternalActorRef = ???
+    private[akka] def isTerminated = ???
+    def isLocal: Boolean = ???
+    def !(message: Any)(implicit sender: ActorRef): Unit = ???
+
+  }
+
+  def stableDummyActorRef(path: ActorPath)(implicit system: ActorSystem): ActorRef = new StableDummyActorRef(path)
```

Comment:
To be able to test-serialize messages with `ActorRefs` we need this, and then also to somehow replace `system.provider.resolveActorRef(str)` with something that resolves any path into these dummy-refs.

Without this the actor-refs will be to different actual actor refs for each test run (new system) and never be equal.

Reply:
Since that's the ActorRefProvider, we could implement one that is "the testing actor ref provider" and set it for the system that does the serialization checks, and mock it to always return this one; WDYT?

Reply:
Yeah, that's the only way I can figure out. I'll try that.